src/status_im/contexts/wallet/wallet_connect/effects.cljs

```diff
@@ -41,10 +46,47 @@
  :effects.wallet-connect/approve-session
  (fn [{:keys [web3-wallet proposal supported-namespaces on-success on-fail]}]
    (let [{:keys [params id]} proposal
-         approved-namespaces (wallet-connect/build-approved-namespaces params
-                                                                       supported-namespaces)]
+         approved-namespaces (wallet-connect/build-approved-namespaces
+                              params
+                              supported-namespaces)]
      (-> (.approveSession web3-wallet
                           (clj->js {:id         id
                                     :namespaces approved-namespaces}))
          (promesa/then on-success)
          (promesa/catch on-fail)))))
+
+(rf/reg-fx
+ :effects.wallet-connect/sign-message
+ (fn [{:keys [password address data on-success on-fail]}]
+   (-> (promesa/-> {:data     data
+                    :account  address
+                    :password (security/safe-unmask-data password)}
+                   bean/->js
+                   transforms/clj->json
+                   native-module/sign-message
+                   wallet-connect-core/extract-native-call-signature)
+       (promesa/then on-success)
+       (promesa/catch on-fail))))
+
+(rf/reg-fx
+ :effects.wallet-connect/sign-typed-data
```

Comment:
A suggestion similar to another comment I made in this review, that maybe you don't need to use the re-frame machinery at this granularity, that is, you don't need to wrap wallet connect functions and expose them as re-frame effects.

For example, the effect `:effects.wallet-connect/sign-typed-data` is only used by the event `:wallet-connect/respond-sign-typed-data`, and this event is only used by event `:wallet-connect/respond-current-session`. If you explore the idea of making `:wallet-connect/respond-current-session` an effect, you might be able to just not even expose/register the lower-level effect `:effects.wallet-connect/sign-typed-data`.

These changes can quite significantly simplify the code.

Reply:
To put from another angle @clauxx, we don't need re-frame effects and events to design a clean wallet connect API with reusable functions to support future code changes. This coupling creates what seems right now to me, unnecessary complexity, and I see this added complexity in a lot of other events related to wallet code (not in this PR alone), but it's also a problem in other parts of the app (chats, etc). Something I'm trying to be more conscious now.

I'll try to be more practical about this type of feedback by refactoring some things in separate PRs if I can find the time. It would be nice if we can pair in some of them o/ Otherwise I'm getting in the realm of high-level feedback in reviews, which I don't like personally. I don't like to just say "hey, this code could be better".

Thanks!

Reply:
IMO this is the kind of feedback I would like to get and I agree with it, but the catch here for me is that I'd feel safer with the refactor once we get all the use cases in place, so it seems a bit premature. I do agree that it looks more complicated to maintain and too verbose/explicit this way, but it was a conscious decision (maybe wrong) so we don't end up with one large function with conditions in arbitrary places, like it was in the previous wallet connect [implementation](https://github.com/status-im/status-mobile/blob/bd7da02c25e97d87c7779c8d1569f8276c6d935e/src/status_im/wallet_connect/core.cljs#L232). It was a bit difficult to find what's what and how things are signed and handled for different rpc methods, but could be that I went into the opposite extreme 🤷 .

Reply:
> This coupling creates what seems right now to me, unnecessary complexity

Do you mean the coupling of an effect with event that does nothing but calling the effect?

Reply:
> Do you mean the coupling of an effect with event that does nothing but calling the effect?

Also that @clauxx 

I fully understand your decision. It makes sense as well, and it's less risky because we don't have much or any good examples in the code to serve as a guide. Let's come back to this open topic in the future!